python/pyarrow/tests/test_dataset.py

```diff
@@ -583,3 +665,51 @@ def test_filter_implicit_cast(tempdir):
     scanner_builder.filter(filter_)
     result = scanner_builder.finish().to_table()
     assert len(result) == 3
+
+
+def test_multiple_sources(multisourcefs):
+    src1 = ds.source('/plain', filesystem=multisourcefs, format='parquet')
+    src2 = ds.source('/schema', filesystem=multisourcefs, format='parquet',
+                     partitioning=['week', 'color'])
+    src3 = ds.source('/hive', filesystem=multisourcefs, format='parquet',
+                     partitioning='hive')
+
+    assembled = ds.dataset([src1, src2, src3])
+    assert isinstance(assembled, ds.Dataset)
+
+    expected_schema = pa.schema([
+        pa.field('date', pa.date32()),
```

Comment:
btw, you don't need the pa.field(), just a tuple is fine as well

Reply:
I'm just used to it, but updated to the less verbose syntax.